pkg/image/registry/image/strategy.go

```diff
@@ -49,6 +49,12 @@ func (s imageStrategy) PrepareForCreate(ctx apirequest.Context, obj runtime.Obje
 		newImage.DockerImageManifest = ""
 		newImage.DockerImageConfig = ""
 	}
+
+	// bug in image update logic allowed to set arbitrary annotations that would otherwise be rejected by
+	// validation; resolves rhbz#1557607
+	for i := range newImage.Signatures {
+		newImage.Signatures[i].Annotations = map[string]string{}
+	}
```

Comment:
should only need this on prepare for update right?

on a create i think we want to reject the creation w/ the validation failure reason, rather than silently drop the user's metadata.

Reply:
I agree. that is consistent with existing behavior

Reply:
What if somebody wants to import backed up images?